========
Overview
========

MicroExonator is a fully-integrated computational pipeline that allows for systematic de novo discovery and quantification
of microexons using raw RNA-seq data for any organism with a gene annotation. Compared to other available methods MicroExonator
is more sensitive for discovering smaller microexons and it provides higher specificity for all lengths. Moreover, MicroExonator
provides integrated downstream comparative analysis between cell types or tissues using
`Whippet <https://github.com/timbitz/Whippet.jl>`_. (`Sterne-Weiler et al. 2018 <https://doi.org/10.1016/j.molcel.2018.08.018>`_).

MicroExonator pipeline is divided in several modules:
    * Discover
    * Quantification
    * Differential Inclusion
    * Single cell analysis

**Support**

    For questions, ideas, feature requests and potential bug reports submit an issue on our GitHub page or write us at gp7@sanger.ac.uk.

.. toctree::
    :name: MicroExonator-install
    :maxdepth: 1
    :hidden:

    install

.. toctree::
    :name: MicroExonator-setup
    :maxdepth: 1
    :hidden:

    setup

.. toctree::
    :name: MicroExonator-discovery-and-quantification
    :maxdepth: 3
    :hidden:

    discovery_and_quantification

.. toctree::
    :name: MicroExonator-differential_inclusion_analysis
    :maxdepth: 3
    :hidden:

    differential_inclusion_analysis

.. toctree::
    :name: MicroExonator-single_cell_analysis
    :maxdepth: 3
    :hidden:

    single_cell_analysis

.. toctree::
    :name: MicroExonator-Licence
    :maxdepth: 1
    :hidden:

    licence

.. toctree::
    :name: MicroExonator-Support
    :maxdepth: 1
    :hidden:

    support
