

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Advanced exercises &mdash; NGS2021_AS  documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Detection of differentially included nodes" href="detection_of_differentially_included_nodes.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> NGS2021_AS
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setup</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install_whippet.html">Whippet: a fast and accurate tool for AS quantification</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="indexing_and_quantifying.html">Transcriptome guided quantification of isoforms</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="detection_of_differentially_included_nodes.html">Detection of differentially included nodes</a></li>
</ul>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Advanced exercises</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">NGS2021_AS</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Advanced exercises</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/advanced_exercises.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="advanced-exercises">
<h1>Advanced exercises<a class="headerlink" href="#advanced-exercises" title="Permalink to this headline">¶</a></h1>
<p>The number of genes that undergoes through alternative splicing is known to be substantially higher than invertebrates or other metazoans. Moreover, the biggest set of regulated AS events are associated with neuronal tissues and are particularly relevant during development of the brain. Therefore, you and your bioinformatician friend were excited to try out Whippet with some of the data that is available at <a class="reference external" href="https://www.encodeproject.org">ENCODE</a>. You decided that would be interesting to process RNA-seq samples from midbrain to detect alternative splicing events between E10.5 and E14.5. To this end,  your bioinformatician friend wrote the following script in bash:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="ch">#!/bin/bash</span>

wget https://www.encodeproject.org/files/ENCFF099UUS/@@download/ENCFF099UUS.fastq.gz -O midbrain_E10.5_rep1_L01.fastq.gz
wget https://www.encodeproject.org/files/ENCFF059FUK/@@download/ENCFF059FUK.fastq.gz -O midbrain_E10.5_rep1_L02.fastq.gz
wget https://www.encodeproject.org/files/ENCFF052VGB/@@download/ENCFF052VGB.fastq.gz -O midbrain_E10.5_rep2_L01.fastq.gz
wget https://www.encodeproject.org/files/ENCFF819ZTA/@@download/ENCFF819ZTA.fastq.gz -O midbrain_E10.5_rep2_L02.fastq.gz

wget https://www.encodeproject.org/files/ENCFF156BSL/@@download/ENCFF156BSL.fastq.gz -O midbrain_E14.5_rep1_L01.fastq.gz
wget https://www.encodeproject.org/files/ENCFF421QJA/@@download/ENCFF421QJA.fastq.gz -O midbrain_E14.5_rep1_L02.fastq.gz
wget https://www.encodeproject.org/files/ENCFF499UQZ/@@download/ENCFF499UQZ.fastq.gz -O midbrain_E14.5_rep2_L01.fastq.gz
wget https://www.encodeproject.org/files/ENCFF093YSD/@@download/ENCFF093YSD.fastq.gz -O midbrain_E14.5_rep2_L02.fastq.gz

wget https://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/mm10.fa.gz -O  mm10.fa.gz
wget https://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/genes/mm10.ensGene.gtf.gz -O mm10.ensGene.gtf.gz

julia /home/gp7/miniconda/envs/julia_0.6.1/share/julia/site/v0.6/Whippet/bin/whippet-index.jl --fasta mm10.fa.gz --gtf mm10.ensGene.gtf.gz --index mm10.index

julia /home/gp7/miniconda/envs/julia_0.6.1/share/julia/site/v0.6/Whippet/bin/whippet-quant.jl &lt;<span class="o">(</span> cat  midbrain_E10.5_rep1*.fastq.gz<span class="o">)</span> --force-gz  -x mm10.index.jls -o midbrain_E10.5_rep1
julia /home/gp7/miniconda/envs/julia_0.6.1/share/julia/site/v0.6/Whippet/bin/whippet-quant.jl &lt;<span class="o">(</span> cat  midbrain_E10.5_rep2*.fastq.gz<span class="o">)</span> --force-gz  -x mm10.index.jls -o midbrain_E10.5_rep2
julia /home/gp7/miniconda/envs/julia_0.6.1/share/julia/site/v0.6/Whippet/bin/whippet-quant.jl &lt;<span class="o">(</span> cat  midbrain_E14.5_rep1*.fastq.gz<span class="o">)</span> --force-gz  -x mm10.index.jls -o midbrain_E14.5_rep1
julia /home/gp7/miniconda/envs/julia_0.6.1/share/julia/site/v0.6/Whippet/bin/whippet-quant.jl &lt;<span class="o">(</span> cat  midbrain_E14.5_rep2*.fastq.gz<span class="o">)</span> --force-gz  -x mm10.index.jls -o midbrain_E14.5_rep2
</pre></div>
</div>
<p>However, as most of bioinformaticians, he was a bit lazy to finish the job and he asked you to do it instead! He sent you the <code class="docutils literal notranslate"><span class="pre">.psi.gz</span></code> he obtained. You can find these files at <code class="docutils literal notranslate"><span class="pre">Data/Advanced/</span></code>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As a general rule <code class="docutils literal notranslate"><span class="pre">rep</span></code> indicates biological replicates and <code class="docutils literal notranslate"><span class="pre">L</span></code> technical replicates, which are samples that were extracted from the same animal but sequenced on different runs. Thus, on this script technical replicates that correspond to the same biological sample are pooled during the quantification step (check <code class="docutils literal notranslate"><span class="pre">--force-gz</span></code> parameter on Whippet’s documentation)</p>
</div>
<div class="admonition-final-challenge admonition">
<p class="admonition-title">FINAL CHALLENGE</p>
<p>Take back control of this splicing project and finish it up by taking the following steps:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Run the last necessary command to detect the alternatively spliced events associated to the embryonic development of mouse midbrain (between E10.5 and E14.E)</p></li>
<li><p>Use your bash skills to calculate how many differentially spliced nodes you detected between these conditions. How many of each splicing node type did you detect?</p></li>
<li><p>Run the <code class="docutils literal notranslate"><span class="pre">Data/Advanced/download_bam.sh</span></code> script to get at least one <code class="docutils literal notranslate"><span class="pre">BAM</span></code> file representing each embryonic stage and use these to generate sashimi plots. Find some examples to show in your presentation.</p></li>
<li><p>Generate a non-redundant list of genes with highly variable Core Exon nodes (CE) that has an absolute <code class="docutils literal notranslate"><span class="pre">DeltaPsi</span></code> value of at least 0.3 and <code class="docutils literal notranslate"><span class="pre">Probability</span></code> greater than 0.9. Use this list to perform <a class="reference external" href="http://bioinformatics.sdstate.edu/go/">gene ontology analyses</a> and protein-protein interaction networks analyses with <a class="reference external" href="https://string-db.org/">STRING</a> (if you reach this point please seek the attention of the instructor in charge of this project).</p></li>
</ol>
</div></blockquote>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="detection_of_differentially_included_nodes.html" class="btn btn-neutral float-left" title="Detection of differentially included nodes" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2021, Guillermo E. Parada

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>