

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Single cell analysis &mdash; NGS2021_AS  documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Discovery and Quantification" href="splicing_node_quantification.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> NGS2021_AS
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setup</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install_whippet.html">Whippet: a fast and accurate tool for AS quantification</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="indexing_transcriptome.html">Differential inclusion analysis</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="splicing_node_quantification.html">Discovery and Quantification</a></li>
</ul>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Single cell analysis</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#configuration">Configuration</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#run-metadata">run_metadata</a></li>
<li class="toctree-l3"><a class="reference internal" href="#optional-configuration">Optional Configuration</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#run">Run</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#unpooled-quantification-optional">Unpooled quantification (optional)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output">Output</a></li>
<li class="toctree-l2"><a class="reference internal" href="#visualization">Visualization</a></li>
</ul>
</li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">NGS2021_AS</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Single cell analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/advanced_exercises.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="single-cell-analysis">
<h1>Single cell analysis<a class="headerlink" href="#single-cell-analysis" title="Permalink to this headline">¶</a></h1>
<p>On this secction we describe how to perform analysis of single-cell RNA-seq data to quantify microexons across populations of cells and alternative splicing events across them. Since single cell experiments usually provides a shallow sequencing depth for each cell, we have developed a pseudo-pooling strategy to assess differential inclusion of microexons and other types of alternative splicing events across defined groups of cells (normally corresponding to cell-types previously defined by gene expresion profiles). We named this single cell analysis module <code class="docutils literal notranslate"><span class="pre">snakepool</span></code> an on this secction we describe how to use it.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Before using this module you must follow the same installation instructions decrived in <span class="xref std std-doc">differential_inclusion_analysis</span> secction.</p>
</div>
<div class="section" id="configuration">
<h2>Configuration<a class="headerlink" href="#configuration" title="Permalink to this headline">¶</a></h2>
<p>To run this secction the following parameters needs to be incorporated at <code class="docutils literal notranslate"><span class="pre">config.yaml</span></code>.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Single_Cell : T
cluster_metadata : /lustre/scratch117/cellgen/team218/gp7/Micro-exons/Runs/Paper/MicroExonator/Whippet/Tasic_clustering.txt
cluster_name : broad_type
file_basename : Run_s
cdf_t : <span class="m">0</span>.8
min_p_mean : <span class="m">0</span>.9
min_delta : <span class="m">0</span>.1
min_rep : <span class="m">25</span>
run_metadata : /lustre/scratch117/cellgen/team218/gp7/Micro-exons/Runs/Paper/MicroExonator/Whippet/Tasic_run.txt
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">Single_Cell</span></code> correspond to an optiona parameter that needs to be set as <code class="docutils literal notranslate"><span class="pre">T</span></code> in order to run <code class="docutils literal notranslate"><span class="pre">snakepool</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cluster_metadata</span></code> must indicate the path of a tabular separated file that contain at least two colums to indicate the cluster and file base names. This file must have the first row as header.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cluster_name</span></code> indicate the name of the column, inside <code class="docutils literal notranslate"><span class="pre">cluster_metadata</span></code>, which has cluster name information.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">file_baseame</span></code> indicate the name of the column, inside <code class="docutils literal notranslate"><span class="pre">cluster_metadata</span></code>, which has the sample names. These needs to match with sample names defined on the input files (See <a class="reference internal" href="setup.html"><span class="doc">Setup</span></a>)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">cdf_t</span></code> parameter set a theshold to run a <a class="reference external" href="https://en.wikipedia.org/wiki/Cumulative_distribution_function">Cumulative distribution function</a> over the resultant Probability values obtained for each node across comutational replicates, asuming these fit a <a class="reference external" href="https://en.wikipedia.org/wiki/Beta_distribution">beta distribution</a>. Which in practical terms can be considered as a user-defined threshold (between 0.5 and 1) to calculate a p-value assosiated to node’s probability of differential inclusion across computational replicates.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">min_p_mean</span></code> corresponds to a threshold of mean probability values across computational samples to define a node as differentially included across the comparing cell-types.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">min_p_delta</span></code> corresponds to a threshold of mean delta PSI values across computational samples to define a node as differentially included across the comparing cell-types.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">min_rep</span></code> minimun set of computational replicates that can be considered to define a node as differentially included across the comparing cell-types. This parameter is relevant because when nodes have limmited read coverage across cells, only some few computational replictates could enable quantitative alternative splicing analyses, leading to unreliable assesment of its alternative inclusion. The number of compuational repeats is defined for each sample inside <cite>run_metadata</cite>. We recomend to set this value to at least the half of the compuational replicates that are scheduled to run by the user.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">run_metadata</span></code> indicates the path of a tabulat separated file which contain information about user-defined comparisons across cell-types. Additional information about this file can be found bellow.</p></li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If <code class="docutils literal notranslate"><span class="pre">Single_Cell</span></code> is set to <code class="docutils literal notranslate"><span class="pre">T</span></code> all the other parameters listed above will become compulsory. Thus, you should only activate this module if you have the required parameters on <code class="docutils literal notranslate"><span class="pre">config.yaml</span></code>.</p>
</div>
<div class="section" id="run-metadata">
<h3>run_metadata<a class="headerlink" href="#run-metadata" title="Permalink to this headline">¶</a></h3>
<p>The file indicated by <code class="docutils literal notranslate"><span class="pre">run_metadata</span></code> must be a tabular separated file containing the following columns:</p>
<table class="docutils align-default" id="id2">
<caption><span class="caption-text"><strong>run_metadata.tsv</strong></span><a class="headerlink" href="#id2" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Column</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Compare_ID</p></td>
<td><p>User-defined name for scheduled comparions</p></td>
</tr>
<tr class="row-odd"><td><p>A.cluster_names</p></td>
<td><p>Comma-separated list of cell-types to be concider as part of <cite>sample group A</cite></p></td>
</tr>
<tr class="row-even"><td><p>A.number_of_pools</p></td>
<td><p>Number of pseudo-bulk to be generated for <cite>sample group A</cite></p></td>
</tr>
<tr class="row-odd"><td><p>B.cluster_names</p></td>
<td><p>Comma-separated list of cell-types to be concider as part of <cite>sample group B</cite></p></td>
</tr>
<tr class="row-even"><td><p>B.number_of_pools</p></td>
<td><p>Number of pseudo-bulk to be generated for <cite>sample group B</cite></p></td>
</tr>
<tr class="row-odd"><td><p>Repeat</p></td>
<td><p>Node type. For more information visit <a class="reference external" href="https://github.com/timbitz/Whippet.jl#output-formats">Whippet’s GitHub page</a>.</p></td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The order of the columns is not relevant, however the column names must correspond to the ones indicated above. Additional columns with other names will be ignored.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>We recomend to consider <code class="docutils literal notranslate"><span class="pre">A.number_of_pools</span></code> and <code class="docutils literal notranslate"><span class="pre">B.number_of_pools</span></code> values that ensure that at least five cells are merged within each pseudo-bulk pool. We also suggest 10 as a minimun value of <code class="docutils literal notranslate"><span class="pre">Repeat</span></code>, higher values will enable better estimation of parameters to fit the resultant probabilities into beta distribution models.</p>
</div>
</div>
<div class="section" id="optional-configuration">
<h3>Optional Configuration<a class="headerlink" href="#optional-configuration" title="Permalink to this headline">¶</a></h3>
<p>The following parameter are optionals to be defined inside <code class="docutils literal notranslate"><span class="pre">config.yaml</span></code> file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>seed : <span class="m">123</span>
Only_snakepool : T
Get_Bamfiles : T
</pre></div>
</div>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">seed</span></code> define a specific seed for pseudo number geration. This number influence the arrangement cells into the corresponding pseudo-bulks. Mataining the same seed ensures reproducibility of the results and prevent snakemake of overwrite completed results.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Only_snakepool</span></code> is a bolean variable that if its defined as <code class="docutils literal notranslate"><span class="pre">T</span></code> it will force MicroExonator to skip Disovery and Quantification modules. This mode is useful for users who are only interested to find alterantive splicing events from splicing nodes that can be extracted from the annotation.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">Get_Bamfiles</span></code> correspond to a bolean variable that if its defined as <code class="docutils literal notranslate"><span class="pre">T</span></code> enable the generation of BAM files that can be used for visualization purposes.</p></li>
</ul>
</div>
</div>
<div class="section" id="run">
<h2>Run<a class="headerlink" href="#run" title="Permalink to this headline">¶</a></h2>
<p>After setting up all the files described above, this single cell analysis module can be run by adding <cite>snakepool</cite> as target for snakemake:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake -s MicroExonator.smk  --cluster-config cluster.json --cluster <span class="o">{</span>cluster system params<span class="o">}</span> --use-conda -k  -j <span class="o">{</span>number of parallel jobs<span class="o">}</span> snakepool
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>It is allways a good idea to use <code class="docutils literal notranslate"><span class="pre">-np</span></code> to execute an snakemake <code class="docutils literal notranslate"><span class="pre">dry-run</span></code> before submiting a large set of jobs.</p>
</div>
<div class="section" id="unpooled-quantification-optional">
<h3>Unpooled quantification (optional)<a class="headerlink" href="#unpooled-quantification-optional" title="Permalink to this headline">¶</a></h3>
<p>In order to generate PSI quantification files at the single cell level (as opposed to pseudo-bulks), you can run MicroExonator with <code class="docutils literal notranslate"><span class="pre">quant_unpool_single_cell</span></code> as a target for snakemake. By doing this <cite>.psi.gz</cite> files will be generated at <code class="docutils literal notranslate"><span class="pre">Whippet/Quant/Single_Cell/Unpooled/</span></code> folder:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake -s MicroExonator.smk  --cluster-config cluster.json --cluster <span class="o">{</span>cluster system params<span class="o">}</span> --use-conda -k  -j <span class="o">{</span>number of parallel jobs<span class="o">}</span> quant_unpool_single_cell
</pre></div>
</div>
<p>This can enable users do run custom downstream analysis over alternative splicing quantification files generated for every cell by separated.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Only FASTQ files from cells annotated on <code class="docutils literal notranslate"><span class="pre">cluster_metadata</span></code> file will be processed.</p>
</div>
</div>
</div>
<div class="section" id="output">
<h2>Output<a class="headerlink" href="#output" title="Permalink to this headline">¶</a></h2>
<p>Direct results from <cite>whippet-delta</cite> for every comparison across each computational replicate can be found at <cite>Whippet/Delta/Single_Cell/</cite>. Integrated results for each comparion can be found at <code class="docutils literal notranslate"><span class="pre">Whippet/Delta/Single_Cell/Sig_nodes</span></code>, these resuls are structured as follow:</p>
<table class="docutils align-default" id="id3">
<caption><span class="caption-text"><strong>all_nodes.microexons.txt</strong></span><a class="headerlink" href="#id3" title="Permalink to this table">¶</a></caption>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>Column</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Gene</p></td>
<td><p>Gene ID</p></td>
</tr>
<tr class="row-odd"><td><p>Node</p></td>
<td><p>Node number inside the gene</p></td>
</tr>
<tr class="row-even"><td><p>Coord</p></td>
<td><p>Node coordinate</p></td>
</tr>
<tr class="row-odd"><td><p>Strand</p></td>
<td><p>Plus or minus strand</p></td>
</tr>
<tr class="row-even"><td><p>Type</p></td>
<td><p>Node type. For more information visit <a class="reference external" href="https://github.com/timbitz/Whippet.jl#output-formats">Whippet’s GitHub page</a>.</p></td>
</tr>
<tr class="row-odd"><td><p>Psi_A.mean</p></td>
<td><p>Mean PSI values for group <code class="docutils literal notranslate"><span class="pre">cluster</span> <span class="pre">A</span></code> across computational replicates.</p></td>
</tr>
<tr class="row-even"><td><p>Psi_B.mean</p></td>
<td><p>Mean PSI values for group <code class="docutils literal notranslate"><span class="pre">cluster</span> <span class="pre">B</span></code> across computational replicates.</p></td>
</tr>
<tr class="row-odd"><td><p>DeltaPsi.mean</p></td>
<td><p>Mean DeltaPsi values obtained across computationa replicates.</p></td>
</tr>
<tr class="row-even"><td><p>DeltaPsi.sd</p></td>
<td><p>Standar deviation of DeltaPsi values obtained across computationa replicates.</p></td>
</tr>
<tr class="row-odd"><td><p>Probability.mean</p></td>
<td><p>Mean probability of differential inclusion obtained across computationa replicates.</p></td>
</tr>
<tr class="row-even"><td><p>Probability.var</p></td>
<td><p>Variance of probability across computationa replicates.</p></td>
</tr>
<tr class="row-odd"><td><p>N.detected.reps</p></td>
<td><p>Number of replicates in which the differential inclusion could be assessed.</p></td>
</tr>
<tr class="row-even"><td><p>cdf.beta</p></td>
<td><p>p-value of being above the used defined probability threshold <code class="docutils literal notranslate"><span class="pre">cdf_t</span></code></p></td>
</tr>
<tr class="row-odd"><td><p>is.diff</p></td>
<td><p>Bolean variable defining wheather the node was differentially included accoding to the user-defined criteria (<code class="docutils literal notranslate"><span class="pre">min_rep</span></code>, <code class="docutils literal notranslate"><span class="pre">min_p_mean</span></code> and <code class="docutils literal notranslate"><span class="pre">min_delta</span></code>)</p></td>
</tr>
<tr class="row-even"><td><p>microexon_ID</p></td>
<td><p>Microexon ID based on its genomic coodinates.</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="visualization">
<h2>Visualization<a class="headerlink" href="#visualization" title="Permalink to this headline">¶</a></h2>
<p>In order to visualize the results, you need to instruct <code class="docutils literal notranslate"><span class="pre">whippet-quant</span></code> to generate SAM files by incorporating the following parameter with a <code class="docutils literal notranslate"><span class="pre">True</span></code> value inside <code class="docutils literal notranslate"><span class="pre">config.yaml</span></code>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Get_Bamfiles : T
</pre></div>
</div>
<p>Samfiles are further converted to BAM files and corresponding index files are genrated to enable their visualization. To generate these BAMs <code class="docutils literal notranslate"><span class="pre">cluster_bams</span></code> needs to be defined as an snakemake target:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>snakemake -s MicroExonator.smk  --cluster-config cluster.json --cluster <span class="o">{</span>cluster system params<span class="o">}</span> --use-conda -k  -j <span class="o">{</span>number of parallel jobs<span class="o">}</span> cluster_bams
</pre></div>
</div>
<p>A BAM file will be generated for every cell-type defined at <code class="docutils literal notranslate"><span class="pre">cluster_metadata</span></code> file. Given the coodinates of differentially included splicing nodes and the correspondig BAM files, sashimi plots can be generated by using tools such as <a class="reference external" href="https://github.com/guigolab/ggsashimi">ggsashimi</a> or <a class="reference external" href="http://software.broadinstitute.org/software/igv/">IGV</a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="splicing_node_quantification.html" class="btn btn-neutral float-left" title="Discovery and Quantification" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2021, Guillermo E. Parada

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>