

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Setup &mdash; NGS2021_AS  documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Whippet: a fast and accurate tool for AS quantification" href="install_whippet.html" />
    <link rel="prev" title="Using RNA-seq to quantify alternative splicing" href="index.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> NGS2021_AS
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Setup</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#installing-conda">Installing Conda</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installing-software-dependencies">Installing software dependencies</a></li>
<li class="toctree-l2"><a class="reference internal" href="#downloading-the-data-and-understanding-the-study-design">Downloading the data and understanding the study design</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#cloning-github-repository-of-this-project">Cloning github repository of this project</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#exercise-1">Exercise 1</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#study-design">Study design</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#exercise-2">Exercise 2</a></li>
<li class="toctree-l4"><a class="reference internal" href="#exercise-3">Exercise 3</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="install_whippet.html">Whippet: a fast and accurate tool for AS quantification</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="indexing_and_quantifying.html">Transcriptome guided quantification of isoforms</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="detection_of_differentially_included_nodes.html">Detection of differentially included nodes</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="advanced_exercises.html">Advanced exercises</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">NGS2021_AS</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Setup</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/setup.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="setup">
<h1>Setup<a class="headerlink" href="#setup" title="Permalink to this headline">¶</a></h1>
<p>The life of the bioinformatician is not easy as some may think. A big part of the time they are trying to install software and reproduce other researcher methods. In this section, you will download all the software and necessary to data to conduct alternative splicing analysis using RNA-seq. To start with this project its necessary to acquire all the software and the data. The following steps will enable us to do this.</p>
<div class="section" id="installing-conda">
<h2>Installing Conda<a class="headerlink" href="#installing-conda" title="Permalink to this headline">¶</a></h2>
<p>Conda is an open-source package management system which enables the installation of a variety of different computational tools. To install conda, please follow this <a class="reference external" href="https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh">link</a>. and download it inside your <code class="docutils literal notranslate"><span class="pre">~/Downloads</span></code> folder. To start installing conda open the terminal and execute the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>bash ~/Downloads/Miniconda3-latest-Linux-x86_64.sh
</pre></div>
</div>
<p>Follow the instructions displayed in the command line by pressing the <code class="docutils literal notranslate"><span class="pre">Enter</span></code> key and writing <code class="docutils literal notranslate"><span class="pre">yes</span></code> when required by the installer. If the installation is successful, you will receive the following message in the prompt.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>Thank you <span class="k">for</span> installing Miniconda2!
</pre></div>
</div>
<p>Moreover, if you list content inside the home directory, you will see a <code class="docutils literal notranslate"><span class="pre">miniconda2</span></code> folder. To make sure conda has the rights to write files on this directory and install software, please insert the following commands:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sudo chmod <span class="m">755</span> -R /home/miniconda2
sudo chmod <span class="m">755</span> -R /home/manager
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The command above will require your password, which is the same than your user name. To check your user name insert <code class="docutils literal notranslate"><span class="pre">whoami</span></code> in the terminal.</p>
</div>
<p>Finally, to test everything in place execute the following command, which will instruct conda to update itself:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda update -n base -c defaults conda
</pre></div>
</div>
</div>
<div class="section" id="installing-software-dependencies">
<h2>Installing software dependencies<a class="headerlink" href="#installing-software-dependencies" title="Permalink to this headline">¶</a></h2>
<p>To install the software necessary for this project, we will first create a virtual environment, and we will install the software dependencies within this virtual environment. This will avoid interfering with the other software installed in the VM. To create a virtual environment called <code class="docutils literal notranslate"><span class="pre">project</span></code> that contain all the necessary software for this project please execute the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda create --name project sra-tools <span class="nv">julia</span><span class="o">=</span><span class="m">0</span>.6.1<span class="o">=</span>blas_openblas_200 gcc_linux-64
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This command may take some minutes to resolve the dependencies, when is finished correctly, it will require you to accept the changes by inserting the <code class="docutils literal notranslate"><span class="pre">y</span></code> key.</p>
</div>
<p>To execute any of the commands related to this project, we first need to activate the environment, <code class="docutils literal notranslate"><span class="pre">project</span></code>, that we just created.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda activate project
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>If you open a new terminal this environment will not be active by default, therefore would not be able to access the required software unless you activate the <code class="docutils literal notranslate"><span class="pre">project</span></code> environment again. Moreover, to inactivate the environment you can use the <cite>deactivate</cite> conda command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda deactivate project
</pre></div>
</div>
</div>
</div>
<div class="section" id="downloading-the-data-and-understanding-the-study-design">
<h2>Downloading the data and understanding the study design<a class="headerlink" href="#downloading-the-data-and-understanding-the-study-design" title="Permalink to this headline">¶</a></h2>
<p>Before starting developing this project we will need to download all the the necessary data. Similarly to the sofware we installed, the data is all available online.</p>
<div class="section" id="cloning-github-repository-of-this-project">
<h3>Cloning github repository of this project<a class="headerlink" href="#cloning-github-repository-of-this-project" title="Permalink to this headline">¶</a></h3>
<p>The first step of starting getting the data is to cloning the git repository of this project. To this end, change the directory to the course’s folder (using <code class="docutils literal notranslate"><span class="pre">cd</span></code> command) and copy the following command in the terminal:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>git clone https://github.com/geparada/NGS_2021_AS
</pre></div>
</div>
<p>This command will make a clone of the repository at your local computer. After the command finish running, a new folder (called <code class="docutils literal notranslate"><span class="pre">NGS_2021_AS</span></code>) will be created at the directory where you executed the command.</p>
<div class="admonition-do-you-want-to-know-more admonition">
<p class="admonition-title">Do you want to know more?</p>
<p>This repository also include the documentation you are currently reading. For more information how this was made visit this <a class="reference external" href="https://docs.readthedocs.io/en/stable/intro/getting-started-with-sphinx.html">link</a>.</p>
</div>
<p>To access data from the <code class="docutils literal notranslate"><span class="pre">NCBI</span> <span class="pre">archive</span></code>, we are going to use fastq-dump, which is part of <a class="reference external" href="https://ncbi.github.io/sra-tools/">SRA-Tools</a>. To start efficiently getting the data we are going to execute a <cite>bash</cite> script which will execute fastq-dump to download some RNA-seq samples. To run this script, change directory (using <code class="docutils literal notranslate"><span class="pre">cd</span></code> command) to <code class="docutils literal notranslate"><span class="pre">Data/samples/</span></code> and execute the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>bash download.sh <span class="p">&amp;</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This command will download five <code class="docutils literal notranslate"><span class="pre">fastq.gz</span></code> files. The character <code class="docutils literal notranslate"><span class="pre">&amp;</span></code> will enable us to keep running this command in the background. Depending on the internet connection this may take between 20-50 minutes. To use <cite>ls -lht</cite> to check the progress in that <code class="docutils literal notranslate"><span class="pre">Data/samples/</span></code>.</p>
</div>
<div class="section" id="exercise-1">
<h4>Exercise 1<a class="headerlink" href="#exercise-1" title="Permalink to this headline">¶</a></h4>
<p>Take a look inside the <code class="docutils literal notranslate"><span class="pre">NGS_2021_AS</span></code> folder and investigate the commands used.
A) Go to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra">SRA</a> and which project the data you are downloading is coming from.
identify the project this data is coming from. Which organism are they coming from?</p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>Identify the accession code of one of the files and use the search button.</p>
</div>
<ol class="upperalpha simple" start="2">
<li><p>Check the metadata of the study and identify the experimental condition of every sample you are currently downloading. Using fastq-dump get the extra</p></li>
</ol>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>Check the Study information and click on <code class="docutils literal notranslate"><span class="pre">All</span> <span class="pre">runs</span></code></p>
</div>
</div>
</div>
<div class="section" id="study-design">
<h3>Study design<a class="headerlink" href="#study-design" title="Permalink to this headline">¶</a></h3>
<p>We are going to first study the effects of TBPH lost of function (LoF) over alternative splicing. For more details about the study please check the information of the study and its corresponding publication.</p>
<div class="section" id="exercise-2">
<h4>Exercise 2<a class="headerlink" href="#exercise-2" title="Permalink to this headline">¶</a></h4>
<p>Ops! the bioinformatician who wrote the script forgot to include one sample to study the effect of TBPH knockout! Before continue please download this file to <code class="docutils literal notranslate"><span class="pre">Data/samples</span></code></p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>Check <code class="docutils literal notranslate"><span class="pre">Data/samples/download.sh</span></code> file and the run metadata available on <code class="docutils literal notranslate"><span class="pre">SRA</span></code>.</p>
</div>
<div class="admonition danger">
<p class="admonition-title">Danger</p>
<p>Do not execute <code class="docutils literal notranslate"><span class="pre">data/samples/download.sh</span></code> again, otherwise you will re-write the downloaded data. But if this happened, do not panic. Re-withing files is a mistake that is far more frequent that one would expect.</p>
</div>
</div>
<div class="section" id="exercise-3">
<h4>Exercise 3<a class="headerlink" href="#exercise-3" title="Permalink to this headline">¶</a></h4>
<p>To make sure you got the final result, your collaborator sent you the expected sizes from the files you are downloading.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>total 2728888
-rw-r--r--  1 gp7  1535   221M 01 Jan 14:14 SRR363940.fastq.gz
-rw-r--r--  1 gp7  1535   330M 01 Jan 14:09 SRR363938.fastq.gz
-rw-r--r--  1 gp7  1535   155M 01 Jan 14:02 SRR363935.fastq.gz
-rw-r--r--  1 gp7  1535   226M 01 Jan 13:59 SRR363932.fastq.gz
-rw-r--r--  1 gp7  1535   133M 01 Jan 13:54 SRR363931.fastq.gz
-rw-r--r--  1 gp7  1535   240M 01 Jan 13:52 SRR363927.fastq.gz
-rw-r--r--  1 gp7  1535   180B 01 Jan 13:41 download.sh
</pre></div>
</div>
<p>Which command do you think your collaborator used to generate this? Use the same command to check if you were able to download the samples properly. If you are still downloading them, please continue with the following modules.</p>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="install_whippet.html" class="btn btn-neutral float-right" title="Whippet: a fast and accurate tool for AS quantification" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="index.html" class="btn btn-neutral float-left" title="Using RNA-seq to quantify alternative splicing" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2021, Guillermo E. Parada

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>